ktor-server/ktor-server-plugins/ktor-server-auth/jvm/src/io/ktor/server/auth/OAuth.kt

```diff
@@ -161,15 +196,31 @@ public sealed class OAuthAccessTokenResponse : Principal {
      * @property tokenType OAuth2 token type (usually Bearer)
      * @property expiresIn token expiration timestamp
      * @property refreshToken to be used to refresh access token after expiration
+     * @property state generated state used for the OAuth procedure
      * @property extraParameters contains additional parameters provided by the server
      */
     public data class OAuth2(
         val accessToken: String,
         val tokenType: String,
         val expiresIn: Long,
         val refreshToken: String?,
-        val extraParameters: Parameters = Parameters.Empty
-    ) : OAuthAccessTokenResponse()
+        val extraParameters: Parameters = Parameters.Empty,
+    ) : OAuthAccessTokenResponse() {
+
+        public var state: String? = null
```

Comment:
just a question: is it expected, that while `OAuth2` is `data class` this **new** property will not appear in `copy`/`componentN`/`equals`/`hashCode`?

Reply:
It's really difficult to add it without breaking binary compatibility. In practice, I don't expect users to do `copy`/`componentN` on principals. And for equals and hashcode, having `accessToken` there should be enough.